src/Core/Managed/Shared/Extensibility/Metrics/DependencyMetricExtractor.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using System.Collections.Generic;
+
+namespace Microsoft.ApplicationInsights.Extensibility.Metrics
+{
+    public class DependencyMetricExtractor : MetricExtractorTelemetryProcessorBase
+    {
+        private const string Version = "1.0";
+
+        private Metric _dependencyCallsDurationMetric;
+        private Metric _dependencyCallsFailuresMetric;
+
+        public DependencyMetricExtractor(ITelemetryProcessor nextProcessorInPipeline)
```

Comment:
The class and all public items need doc comments (and I'll be quite happy if you also comment the private stuff as well).